inksdk/ink/src/main/java/com/microsoft/device/ink/InkView.kt

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+
+package com.microsoft.device.ink
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BlendMode
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.SurfaceTexture
+import android.os.Build
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Surface
+import android.view.TextureView
+import androidx.annotation.RequiresApi
+
+@RequiresApi(Build.VERSION_CODES.Q)
+class InkView constructor(
+    context: Context,
+    attributeSet: AttributeSet
+) :
+    TextureView(context, attributeSet), TextureView.SurfaceTextureListener {
+
+    private var mSurface: Surface? = null
+    private var mSurfaceTexture: SurfaceTexture? = null
+    private var inputManager: InputManager
+    private lateinit var canvasBitmap: Bitmap
+    private lateinit var drawCanvas: Canvas
+    private val currentStrokePaint = Paint()
+    private val strokeList = mutableListOf<InputManager.ExtendedStroke>()
+    private val overridePaint: Paint
+    private val clearPaint: Paint
+
+    // attributes
+    private var enablePressure = false
+    private var minStrokeWidth = 1f
+    private var maxStrokeWidth = 10f
+
+    // properties
+    var color = Color.GRAY
+        set(value) {
+            field = value
+            currentStrokePaint.color = value
+        }
+
+    private var _dynamicPaintHandler: DynamicPaintHandler? = null
+    var dynamicPaintHandler = _dynamicPaintHandler
+        set(value) {
+            _dynamicPaintHandler = value
+            field = value
+        }
+    interface DynamicPaintHandler {
+        fun generatePaintFromPenInfo(penInfo: InputManager.PenInfo): Paint
+    }
+
+    init {
+        isOpaque = false // mae the texture view transparent!
+        // handle attributes
+        context.theme.obtainStyledAttributes(
+            attributeSet,
+            R.styleable.InkView,
+            0, 0
+        ).apply {
+            try {
+                enablePressure = getBoolean(R.styleable.InkView_enable_pressure, enablePressure)
+                color = getColor(R.styleable.InkView_ink_color, color)
+                minStrokeWidth = getFloat(R.styleable.InkView_min_stroke_width, minStrokeWidth)
+                maxStrokeWidth = getFloat(R.styleable.InkView_max_stroke_width, maxStrokeWidth)
+            } finally {
+                recycle()
+            }
+        }
+        this.surfaceTextureListener = this
```

Comment:
Why you need this? Because this class implements the SurfaceTextureListener, when you need to call some functions from this interface, you can use directly "this.someFunction()"

Reply:
its not me. its the system that calls onSurfaceTextureAvailable when the surface is ready. and then I can lock it and paint on it.